invenio_indexer/utils.py

```diff
@@ -61,14 +61,14 @@ def default_record_to_index(record):
 
     index, doc_type = schema_to_index(schema, index_names=index_names)
 
-    if not (index and doc_type):
```

Comment:
Shouldn't we keep this to keep the ES6 compatibility?

Reply:
if we keep it, then it won't work for OS2 because everywhere we need `doc_type = None`, so all the indexes will enter this block and index name will be set to record-v1.0.0 (I tested it in ILS)

Reply:
But it will be set to `None` because of the `if ES_VERSION[0] >= 7:` below. This if clause doesn't return, it just sets to default (and then below it's going to be "cleared" to `None`).

Reply:
Sorry had a brainfart, I'm totally wrong, we need this :)

Reply:
yes, if doc_type or index is none (we care about the doc_type)  it sets all the indices names to records-record-v1.0.0, not the respective ones set in the instance (for example we have documents-document-v1.0.0, etc and they are being defaulted to record index due to this if clause. It it not ideal, because the mappings will not work anymore (or indices)

Reply:
I do agree with the logic, I'm not a 100% sure why we actually have `doc_type=None` in OS 2 with the `if` from line 30.

opensearch-py sets the version string to 2.2.0 [here](https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/_version.py#L27) and parses it [here](https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/__init__.py#L41). Unless the _compatibility mode_ messes with this.

Reply:
`invenio-indexer v1.x` is not meant to be used with the revamped `invenio-search` versions that support both ES and OS. Basically what we're doing here is making the `elasticsearch-py` client compatible with an OSv2 cluster w/ compatibility mode enabled.